### PR TITLE
feat: Added Entity Framework Core's schemas

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: Skyra/Database/Models/User.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		public long Money { get; set; } = 0;
+		public uint PointCount { get; set; } = 0;
+		public uint ReputationCount { get; set; } = 0;
+		public string ThemeLevel { get; set; } = "1001";
+		public string ThemeProfile { get; set; } = "0001";
+		public bool DarkTheme { get; set; } = false;
+		public bool ModerationDm { get; set; } = true;
+		public DateTime? NextDaily { get; set; } = null;
+		public DateTime? NextReputation { get; set; } = null;
+		public int Vault { get; set; } = 0;
```
Comment (Tylertron1998) [inline Skyra/Database/Models/User.cs]: can the Vault be negative? If not, consider using an unsigned type - perhaps ushort?

File: Skyra/Database/Models/Guild.cs [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+	{
+		None
+	}
+
+	public enum GuildModerationHardAction
+	{
+		None
+	}
+
+	public abstract class GuildModerationShared
```
Comment (Tylertron1998) [inline Skyra/Database/Models/Guild.cs]: perhaps consider GuildModerationBase as a name - seems to make more sense to me.

File: Skyra/Database/Models/Guild.cs [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+		public GuildRole Role { get; set; }
+		public GuildAutoModeration AutoModeration { get; set; }
+		public GuildSocial Social { get; set; }
+		public GuildAction Action { get; set; }
+		public GuildMusic Music { get; set; }
+	}
+
+	public struct GuildTag
+	{
+		[JsonProperty("name")]
```
Comment (Tylertron1998) [inline Skyra/Database/Models/Guild.cs]: are these JsonPropertyAttributes needed? AFAIK; at least with Newtonsoft.Json - the casing doesn't matter; `name` and `Name` is the same thing to it.

**Review by Tylertron1998 — COMMENTED** [review]
Some questions in review; also 
- why are all of these classes? Can they not be structs? If so, they'd be unmodifiable in efcores cache.
- why does every property have a getter and a setter? Is this because of efcore requiring?

other then this; looks good to me. Good job and well done!

File: Skyra/Database/Models/User.cs (hunk 1 above)
Comment (kyranet) [reply Skyra/Database/Models/User.cs]: Actually that should be `ulong`

**Comment by kyranet** [comment]
- Classes are the way:tm: to make entities in EF, at least, according to [many] ([seriously]) [docs].
- Ditto ^

[many]: https://docs.microsoft.com/en-us/ef/core/modeling/
[seriously]: https://www.learnentityframeworkcore.com/
[docs]: https://www.entityframeworktutorial.net/code-first/simple-code-first-example.aspx

**Comment by Tylertron1998** [comment]
Looks good to me!

**Pull request merged**